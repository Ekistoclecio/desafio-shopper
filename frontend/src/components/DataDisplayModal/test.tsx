import { screen } from '@testing-library/react';

import { DataDisplayModal } from '.';

import { renderWithTheme } from '@/utils/tests/helpers';

describe('<DataDisplayModal />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <DataDisplayModal open={true} handleClose={() => {}} title="Modal de exemplo">
        <p>Conteúdo do modal</p>
      </DataDisplayModal>,
    );

    expect(screen.getByRole('heading', { name: /Modal de exemplo/i })).toBeInTheDocument();
    expect(screen.getByText(/Conteúdo do modal/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call handleClose when close button is clicked', () => {
    const handleClose = jest.fn();

    renderWithTheme(
      <DataDisplayModal open={true} handleClose={handleClose} title="Modal de exemplo">
        <p>Conteúdo do modal</p>
      </DataDisplayModal>,
    );

    const closeButton = screen.getByRole('button', { name: /fechar/i });
    closeButton.click();

    expect(handleClose).toHaveBeenCalled();
  });
});
