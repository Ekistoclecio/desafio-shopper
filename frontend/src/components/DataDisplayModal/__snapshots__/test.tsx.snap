// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataDisplayModal /> should render the heading 1`] = `null`;
